shellwhat/checks.py

```diff
@@ -1,5 +1,49 @@
 import re
 from functools import partial
+from protowhat.checks.check_logic import *
+
+def test_student_typed(state, text, msg="Submission does not contain the code `{}`.", fixed=False):
```

Comment:
Yeah, I need to think about generalizing it, since in `sqlwhat` you can use it after `check_node` etc..

Shouldn't be a problem, but in general, I still need to go carefully over what pieces of sqlwhat's AST Dispatcher to pull into protowhat (so we can also pull `check_node` etc into protowhat).